[fix] 카테고리 삭제 시 연관 투두 자동 삭제 처리

## 🍞 Issue
`Category` 삭제 시, 해당 `Category`를 참조 중인 `Todo` 데이터가 존재할 경우
MySQL의 외래 키 제약 조건(Foreign Key Constraint)으로 인해 삭제가 실패함.

```bash
Caused by: java.sql.SQLIntegrityConstraintViolationException:
Cannot delete or update a parent row: a foreign key constraint fails
(`bbangzip`.`todo`, CONSTRAINT `FKeh6uro943emclp6kr1d4wysmo` FOREIGN KEY (`category_id`) REFERENCES `category` (`id`))
```

## 🥖 Refactor Summary
`Category` 삭제 시 연관된 `Todo`도 함께 삭제되도록 Cascade 설정(`CascadeType.REMOVE`) 추가.  
기획 요구사항에 따라 `Todo`가 남아 있더라도 `Category`를 자유롭게 삭제 가능하도록 수정.


## 🍪 Changes
- [ ] todo가 존재하더라도 category를 삭제할 수 있도록 수정
- [ ] Category 삭제 시 연관된 Todo도 함께 삭제되도록 Cascade 설정 추가

--- a/bbangzip-domain/src/main/java/org/sopt/category/domain/CategoryEntity.java
+++ b/bbangzip-domain/src/main/java/org/sopt/category/domain/CategoryEntity.java
@@ -1,23 +1,16 @@
 package org.sopt.category.domain;
 
-import jakarta.persistence.Column;
-import jakarta.persistence.Entity;
-import jakarta.persistence.EnumType;
-import jakarta.persistence.Enumerated;
-import jakarta.persistence.FetchType;
-import jakarta.persistence.GeneratedValue;
-import jakarta.persistence.GenerationType;
-import jakarta.persistence.Id;
-import jakarta.persistence.JoinColumn;
-import jakarta.persistence.ManyToOne;
-import jakarta.persistence.Table;
+import jakarta.persistence.*;
 import lombok.AccessLevel;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import org.sopt.common.BaseTimeEntity;
+import org.sopt.todo.domain.TodoEntity;
 import org.sopt.user.domain.UserEntity;
 
 import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
 
 import static org.sopt.category.domain.CategoryTableConstants.COLUMN_COLOR;
 import static org.sopt.category.domain.CategoryTableConstants.COLUMN_ID;
@@ -55,6 +48,13 @@ public class CategoryEntity extends BaseTimeEntity {
 
     @Column(name = STOPPED_DATE)
     private LocalDateTime stoppedDate;
+
+    @OneToMany(
+            mappedBy = "category",
+            cascade = CascadeType.REMOVE,
+            orphanRemoval = true
+    )
+    private List<TodoEntity> todos = new ArrayList<>();
 
     @Column(name = COLUMN_ORDER, nullable = false)
     private int order;

